File: Easy/ValidPalindrome.py
# https://leetcode.com/problems/valid-palindrome/
import re 

class Solution(object):
    def isPalindrome(self, s):
        pattern = r'[A-Za-z]'
        string = ""
        for i in s:
            if re.match(pattern, i):
                string += i
            # print(string)
        if string.lower() == string[::-1].lower():
            return True
        return False

s = input()  
a = Solution()
print(a.isPalindrome(s))
# A man, a plan, a canal: Panama

File: Easy/ValidPalindrome.java
package Easy;

import java.util.Scanner;

// # https://leetcode.com/problems/valid-palindrome/
public class ValidPalindrome {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String str = scanner.nextLine();
        scanner.close();
        System.out.println(palindrome(str));
    }

    public static boolean palindrome(String str){
        StringBuilder result = new StringBuilder();
        int i = str.length();
        while(i-->0){
            if (Character.isLetterOrDigit(str.charAt(i))){
                result.append(str.charAt(i));
            }
        }
        System.err.println(result);
        if (result.toString().toLowerCase().equals(result.reverse().toString().toLowerCase())){
            return true;
        }
        return false;
    }
}
// A man, a plan, a canal: Panama